lib/http/HttpServer.ts

```diff
@@ -54,35 +54,43 @@ class HttpServer {
       return;
     }
 
-    let resJson: any;
-    try {
-      switch (req.url) {
-        case '/resolveraiden':
-          resJson = await this.httpService.resolveHashRaiden(reqJson);
-          break;
-        default:
-          res.writeHead(404);
-          res.end();
-          break;
-      }
-      res.writeHead(200, { 'Content-Type': 'application/json' });
-      res.end(JSON.stringify(resJson));
-    } catch (err) {
-      switch (err.code) {
-        case swapErrorCodes.INVALID_RESOLVE_REQUEST:
-        case errorCodes.INVALID_ARGUMENT:
-          res.writeHead(400);
-          res.end(err.message);
-          break;
-        case swapErrorCodes.PAYMENT_HASH_NOT_FOUND:
-          res.writeHead(404);
-          res.end(err.message);
-          break;
-        default:
-          res.writeHead(500);
-          res.end(JSON.stringify(err));
-          break;
-      }
+    switch (req.url) {
+      case '/resolveraiden':
+        try {
+          const resJson = await this.httpService.resolveHashRaiden(reqJson);
+          res.writeHead(200, { 'Content-Type': 'application/json' });
+          res.end(JSON.stringify(resJson));
+        } catch (err) {
+          switch (err.code) {
+            case swapErrorCodes.INVALID_RESOLVE_REQUEST:
+            case errorCodes.INVALID_ARGUMENT:
+              res.writeHead(400);
+              res.end({ message: err.message, retry: false });
+              break;
+            case swapErrorCodes.PAYMENT_HASH_NOT_FOUND:
+              res.writeHead(404);
+              res.end({ message: err.message, retry: false });
+              break;
+            case swapErrorCodes.FINAL_PAYMENT_ERROR:
+            case swapErrorCodes.PAYMENT_REJECTED:
+              res.writeHead(500);
+              res.end({ message: err.message, retry: false });
```

Comment:
What is the called expected to do?  
Are we sure that second leg payment was not done?

Reply:
Yes, although this relies on the logic behind `FINAL_PAYMENT_ERROR` and `PAYMENT_REJECTED` to only be used when we're sure the payment has not and will not go through.